rdt/hyper_transformer.py

```diff
@@ -220,17 +220,37 @@ def get_default_sdtype_transformers(self):
         """
         return self.default_sdtype_transformers
 
-    def update_default_sdtype_transformers(self, new_sdtype_transformers):
-        """Update the ``default_sdtype_transformer`` dict.
+    def update_transformers_by_sdtype(self, sdtype, transformer):
+        """Update the transformers for the specified ``sdtype``.
+
+        Given an ``sdtype`` and a ``transformer``, change all the fields of the ``sdtype``
+        to use the given transformer.
 
         Args:
-            new_sdtype_transformers (dict):
-                Dict mapping sdtypes to the default transformer class or instance to use for
-                them. This dict does not need to contain an entry for every sdtype. It will be
-                used to overwrite the existing defaults. Calling this method will require ``fit``
-                to be run again.
+            sdtype (str):
+                Semantic data type for the transformer.
+            transformer (rdt.transformers.BaseTransformer):
+                Transformer class or instance to be used for the given ``sdtype``.
         """
-        self.default_sdtype_transformers.update(new_sdtype_transformers)
+        self.default_sdtype_transformers[sdtype] = transformer
+        if not self.field_sdtypes:
+            msg = (
+                'Error: Nothing to update. Use the `detect_initial_config` method to '
```

Comment:
I think we actually want to raise an exception here. If they do not have a config, we will not allow them to run these methods